tensorflow_datasets/testing/dataset_builder_testing.py

```diff
@@ -386,6 +389,20 @@ def _download_and_prepare_as_dataset(self, builder):
     with self._subTest("config_description"):
       self._test_description_builder_config(builder)
 
+  def _assertNoFileName(self, builder):
+      original_generate_examples = builder._generate_examples
+
+      def new_generate_examples(self, *args, **kwargs):
+        return original_generate_examples(self, *args, **kwargs)
```

Comment:
You need to add the validation inside the new_generate_examples, by checking the returned value of original_generate_examples